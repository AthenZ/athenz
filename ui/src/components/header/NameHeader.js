/*
 * Copyright The Athenz Authors
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import styled from '@emotion/styled';
import React from 'react';
import Menu from '../denali/Menu/Menu';
import Icon from '../denali/icons/Icon';
import { colors } from '../denali/styles';
import Link from 'next/link';
import { withRouter } from 'next/router';
import PageUtils from '../utils/PageUtils';

const StyledAnchor = styled.a`
    color: #3570f4;
    text-decoration: none;
    cursor: pointer;
`;

const TitleDiv = styled.div`
    font: 600 20px HelveticaNeue-Reg, Helvetica, Arial, sans-serif;
    margin-bottom: 10px;
`;

const MenuDiv = styled.div`
    padding: 5px 10px;
    background-color: black;
    color: white;
    font-size: 12px;
`;

class NameHeader extends React.Component {
    constructor(props) {
        super(props);
    }

    render() {
        const { domain, collectionDetails, collection } = this.props;

        let iconDelegated = (
            <Menu
                placement='bottom-start'
                trigger={
                    <span>
                        <Icon
                            icon={'network'}
                            color={colors.icons}
                            isLink
                            size={'1.25em'}
                            verticalAlign={'text-bottom'}
                        />
                    </span>
                }
            >
                <MenuDiv>Delegated Role</MenuDiv>
            </Menu>
        );

        let iconAudit = (
            <Menu
                placement='bottom-start'
                trigger={
                    <span>
                        <Icon
                            icon={'file-search'}
                            color={colors.icons}
                            isLink
                            size={'1.25em'}
                            verticalAlign={'text-bottom'}
                            onHover={'audit enabled role'}
                        />
                    </span>
                }
            >
                <MenuDiv>Audit Enabled {this.props.category}</MenuDiv>
            </Menu>
        );

        let roleTypeIcon = collectionDetails.trust ? iconDelegated : '';
        let roleAuditIcon = collectionDetails.auditEnabled ? iconAudit : '';

        if (collectionDetails.trust) {
            let deDomain = collectionDetails.trust;
            return (
                <TitleDiv data-testid='collection-name-header'>
                    {roleTypeIcon}
                    {roleAuditIcon}
                    <Link href={PageUtils.rolePage(domain)} passHref legacyBehavior>
                        <StyledAnchor>{domain}</StyledAnchor>
                    </Link>
                    :role.{collection}
                    {' (Delegated to '}
                    <Link href={PageUtils.rolePage(deDomain)} passHref legacyBehavior>
                        <StyledAnchor>{deDomain}</StyledAnchor>
                    </Link>
                    {')'}
                </TitleDiv>
            );
        }
        let link;
        if (this.props.category === 'group') {
            link = (
                <Link href={PageUtils.groupPage(domain)} passHref legacyBehavior>
                    <StyledAnchor>{domain}</StyledAnchor>
                </Link>
            );
        } else if (this.props.category === 'role') {
            link = (
                <Link href={PageUtils.rolePage(domain)} passHref legacyBehavior>
                    <StyledAnchor>{domain}</StyledAnchor>
                </Link>
            );
        } else if (this.props.category === 'policy') {
            link = (
                <Link href={PageUtils.servicePage(domain)} passHref legacyBehavior>
                    <StyledAnchor>{domain}</StyledAnchor>
                </Link>
            );
        } else if (this.props.category === 'service') {
            link = (
                <Link href={PageUtils.servicePage(domain)} passHref legacyBehavior>
                    <StyledAnchor>{domain}</StyledAnchor>
                </Link>
            );
        }

        return (
            <TitleDiv data-testid='collection-name-header'>
                {roleTypeIcon}
                {roleAuditIcon}
                {link}:{this.props.category}.{collection}
            </TitleDiv>
        );
    }
}

export default withRouter(NameHeader);
