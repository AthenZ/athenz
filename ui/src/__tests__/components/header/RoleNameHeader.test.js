/*
 * Copyright The Athenz Authors
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import React from 'react';
import { render } from '@testing-library/react';
import NameHeader from '../../../components/header/NameHeader';

describe('Header', () => {
    it('should render', () => {
        let domain = 'home.mujibur';
        let role = 'admin';
        let roleDetails = {
            trust: null,
            auditEnabled: false,
        };
        const { getByTestId } = render(
            <NameHeader
                showSearch={false}
                domain={domain}
                collection={role}
                collectionDetails={roleDetails}
            />
        );
        const header = getByTestId('collection-name-header');
        expect(header).toMatchSnapshot();
    });
    it('should render audit enabled role', () => {
        let domain = 'home.mujibur';
        let role = 'audit.role';
        let roleDetails = {
            trust: null,
            auditEnabled: true,
        };
        const { getByTestId } = render(
            <NameHeader
                showSearch={false}
                domain={domain}
                collection={role}
                collectionDetails={roleDetails}
            />
        );
        const header = getByTestId('collection-name-header');
        expect(header).toMatchSnapshot();
    });
    it('should render delegated role', () => {
        let domain = 'home.mujibur';
        let role = 'delegated.role';
        let roleDetails = {
            trust: 'home.hga',
            auditEnabled: false,
        };
        const { getByTestId } = render(
            <NameHeader
                showSearch={false}
                domain={domain}
                collection={role}
                collectionDetails={roleDetails}
            />
        );
        const header = getByTestId('collection-name-header');
        expect(header).toMatchSnapshot();
    });
});
