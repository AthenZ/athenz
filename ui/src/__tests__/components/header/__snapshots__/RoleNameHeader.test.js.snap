// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render 1`] = `
.emotion-0 {
  font: 600 20px HelveticaNeue-Reg,Helvetica,Arial,sans-serif;
  margin-bottom: 10px;
}

<div
  class="emotion-0 emotion-1"
  data-testid="collection-name-header"
>
  :
  .
  admin
</div>
`;

exports[`Header should render audit enabled role 1`] = `
.emotion-0 {
  font: 600 20px HelveticaNeue-Reg,Helvetica,Arial,sans-serif;
  margin-bottom: 10px;
}

.emotion-2 {
  fill: #188fff;
  cursor: pointer;
  vertical-align: text-bottom;
}

<div
  class="emotion-0 emotion-1"
  data-testid="collection-name-header"
>
  <span>
    <svg
      class="emotion-2"
      data-testid="icon"
      height="1.25em"
      id=""
      viewBox="0 0 1024 1024"
      width="1.25em"
    >
      <title>
        file-search
      </title>
      <path
        d="M682.667 533.333c0-23.564-19.103-42.667-42.667-42.667v0h-341.333c-23.564 0-42.667 19.103-42.667 42.667s19.103 42.667 42.667 42.667v0h341.333c23.564 0 42.667-19.103 42.667-42.667v0z"
      />
      <path
        d="M298.667 661.333c-23.564 0-42.667 19.103-42.667 42.667s19.103 42.667 42.667 42.667v0h170.667c23.564 0 42.667-19.103 42.667-42.667s-19.103-42.667-42.667-42.667v0z"
      />
      <path
        d="M768 896h-597.333v-725.333h213.333c23.564 0 42.667-19.103 42.667-42.667v0 0c0-23.564-19.103-42.667-42.667-42.667v0h-256c-23.564 0-42.667 19.103-42.667 42.667v0 810.667c0 23.564 19.103 42.667 42.667 42.667v0h682.667c23.564 0 42.667-19.103 42.667-42.667v0-341.333c0-23.564-19.103-42.667-42.667-42.667v0 0c-23.564 0-42.667 19.103-42.667 42.667v0z"
      />
      <path
        d="M968.533 482.133l-108.8-108.8c22.702-32.73 36.267-73.292 36.267-117.023 0-0.109-0-0.218-0-0.328l0 0.017c0-117.821-95.513-213.333-213.333-213.333s-213.333 95.513-213.333 213.333v0c0.259 22.965 4.143 44.937 11.104 65.488l-0.438-1.488h-181.333c-23.564 0-42.667 19.103-42.667 42.667s19.103 42.667 42.667 42.667v0h232.533c38.67 39.114 92.173 63.493 151.371 63.999l0.096 0.001c0.092 0 0.201 0 0.31 0 43.731 0 84.293-13.565 117.709-36.717l-0.686 0.45 108.8 108.8c7.434 8.083 18.061 13.13 29.867 13.13s22.432-5.047 29.84-13.101l0.026-0.029c8.083-7.434 13.13-18.061 13.13-29.867s-5.047-22.432-13.101-29.84l-0.029-0.026zM554.667 256c0-70.692 57.308-128 128-128s128 57.308 128 128c0 70.692-57.308 128-128 128v0c-70.692 0-128-57.308-128-128v0z"
      />
    </svg>
  </span>
  :
  .
  audit.role
</div>
`;

exports[`Header should render delegated role 1`] = `
.emotion-0 {
  font: 600 20px HelveticaNeue-Reg,Helvetica,Arial,sans-serif;
  margin-bottom: 10px;
}

.emotion-2 {
  fill: #188fff;
  cursor: pointer;
  vertical-align: text-bottom;
}

.emotion-3 {
  color: #3570f4;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
}

<div
  class="emotion-0 emotion-1"
  data-testid="collection-name-header"
>
  <span>
    <svg
      class="emotion-2"
      data-testid="icon"
      height="1.25em"
      id=""
      viewBox="0 0 1024 1024"
      width="1.25em"
    >
      <title>
        network
      </title>
      <path
        d="M938.667 682.667h-42.667v-170.667c0-23.564-19.103-42.667-42.667-42.667v0h-298.667v-128h42.667c23.564 0 42.667-19.103 42.667-42.667v0-170.667c0-23.564-19.103-42.667-42.667-42.667v0h-170.667c-23.564 0-42.667 19.103-42.667 42.667v0 170.667c0 23.564 19.103 42.667 42.667 42.667v0h42.667v128h-298.667c-23.564 0-42.667 19.103-42.667 42.667v0 170.667h-42.667c-23.564 0-42.667 19.103-42.667 42.667v0 170.667c0 23.564 19.103 42.667 42.667 42.667v0h170.667c23.564 0 42.667-19.103 42.667-42.667v0-170.667c0-23.564-19.103-42.667-42.667-42.667v0h-42.667v-128h256v128h-42.667c-23.564 0-42.667 19.103-42.667 42.667v0 170.667c0 23.564 19.103 42.667 42.667 42.667v0h170.667c23.564 0 42.667-19.103 42.667-42.667v0-170.667c0-23.564-19.103-42.667-42.667-42.667v0h-42.667v-128h256v128h-42.667c-23.564 0-42.667 19.103-42.667 42.667v0 170.667c0 23.564 19.103 42.667 42.667 42.667v0h170.667c23.564 0 42.667-19.103 42.667-42.667v0-170.667c0-23.564-19.103-42.667-42.667-42.667v0zM213.333 853.333h-85.333v-85.333h85.333zM469.333 170.667h85.333v85.333h-85.333zM554.667 853.333h-85.333v-85.333h85.333zM896 853.333h-85.333v-85.333h85.333z"
      />
    </svg>
  </span>
  <a
    class="emotion-3 emotion-4"
    href="/domain/home.mujibur/role"
  >
    home.mujibur
  </a>
  :role.
  delegated.role
   (Delegated to 
  <a
    class="emotion-3 emotion-4"
    href="/domain/home.hga/role"
  >
    home.hga
  </a>
  )
</div>
`;
